### DISCUSSION TITLE
Remove action help

### DISCUSSION THREAD

**Discussion by bagaze** [ibhagwan/fzf-lua#1577] [Q&A] [answered]
Hello.

I wanted to know if it was possible to hide the **action help string** from the search panel ? I find it a bit distracting.

<img width="493" alt="Screenshot 2024-12-17 at 00 11 04" src="https://github.com/user-attachments/assets/33972578-3d48-4d55-8e86-9b6afd44944b" />

Thanks!

**Comment by ibhagwan** [comment, marked as answer]
You can remove all headers by setting `header = false` under the right picker options or directly in the call `:FzfLua live_grep header=false` or use `no_header_i=true` to remove only the “interactive”  headers.

**Reply by bagaze** [reply]
Thanks!

If set through settings, is it possible to do it globally or do you need to do it picker per picker?

**EDIT: I found the answer in the documentation. You need to set in the `defaults` section**

```
defaults = {
  no_header = true, -- hide grep|cwd header?
},
```

**Reply by milanglacier** [reply]
Setting `no_header_i = true` in the `defaults` is really useful!

Some pickers, like `lsp`, and `dap` does not respect `no_header_i = true` in their picker config, aka some thing like this not work:


```
    dap = {
        no_header_i = true
    }
```

However setting 

```lua
defaults = {
  no_header_i = true, -- hide grep|cwd header?
}
```

effectively disables the headers for all of the pickers.

Beside, I think `header` is a confusing name, because the title of the floating window (if you are using `default-title` preset) can also be referred to as `header`. Maybe `suggestion_tooltip` is a better name?

**Updates**: Okay I got it why it is named as `header`. So I just searched `man fzf`. Because `fzf` called it as `header`, aka you pass `--header '::<A-h> to toggle hidden'` to the fzf process. So `header` might be a proper name if the user are familiar with `fzf` command line app and want to find the equivalent quickly. But this is not an intuitive name for users who are not familiar with `fzf` to find it anyway.

**Reply by ibhagwan** [reply]
> Some pickers, like lsp, and dap does not respect no_header_i = true in their picker config, aka some thing like this not work

I ahould probably do some renaming an clarifications, no_header_i is just the “interactive” headers which are actions the toggle thing / reload / etc.

If you wish to remove all headers simply set `header=false`.

> Beside, I think header is a confusing name, because the title of the floating window (if you are using default-title preset) can also be referred to as header. Maybe suggestion_tooltip is a better name?

That’s the name of the fzf flag responsible for this  `--header`.

**Reply by milanglacier** [reply]
Thanks for the updates for the clarification!

I think if I can just remove all those suggestions tips that should be enough. If there are any headers (actually I haven't see any other headers except the suggestion tips during my usage). I may be interested to see what are those other headers and decide if I want to disable them.

**Reply by milanglacier** [reply]
Beside, `dap` and `lsp` does not respect `header` either, so this also not works:

```lua
    dap = {
        header = false
    }
```

The only way to disable header for them is to set it in `defaults`.

**Reply by ibhagwan** [reply]
> Thanks for the updates for the clarification!
> 
> I think if I can just remove all those suggestions tips that should be enough. If there are any headers (actually I haven't see any other headers except the suggestion tips during my usage). I may be interested to see what are those other headers and decide if I want to disable them.

If you run files on a different folder than the cwd, you’ll get a header note on which is the active cwd, when switching from live grep to grep (fuzzy) you’ll get a header hint on what is the current regex searched for by rg.

> Beside, dap and lsp does not respect header either, so this also not works:
> The only way to disable header for them is to set it in defaults.

Global defaulfs merge into the picker defaulfs so it should work, LSP is a few different categories, lsp is for all locations (refs, defs, etc), lsp.symbols for symbols and lsp.finder for the lsp finder, check out the default options in the README you’ll see it all.

Same for dap, there are sub categories, so you need to use ` dap = { <picker> = { header = false } }`.

**Reply by ibhagwan** [reply]
Actually dap isn’t in the README (too many options there already, need refactoring), see defaults.lua for all categories:
https://github.com/ibhagwan/fzf-lua/blob/34d5e4053a3825ef16a04aa3664dc856c6a13e64/lua/fzf-lua/defaults.lua#L1139-L1170

**Reply by milanglacier** [reply]
Thank you! Man you reply the things insanely promptly, I cannot say enough for my appreciation!

> If you run files on a different folder than the cwd, you’ll get a header note on which is the active cwd, when switching from live grep to grep (fuzzy) you’ll get a header hint on what is the current regex searched for by rg.

This is really useful! I will not disable them.

>  dap = { <picker> = { header = false } }

Thanks for the clarification!

**Comment by bleacheda** [comment]
Hello, on the opposite end I **want** to have the header, especially for live_grep, but when setting it up like so, I have the header appear on everything else *except* live_grep, not sure what I am doing wrong:
```sh
  config = function()
    require("fzf-lua").setup({

      fzf_opts = {
        ["--header"] = ":: <alt-h> to Include hidden files | <alt-i> to Disable .gitignore | <ctrl-g> to Fuzzy Search",
      },

      vim.api.nvim_set_keymap("n", "<leader>ff", ":FzfLua files<CR>", { desc = "fzf files" }),
      vim.api.nvim_set_keymap("n", "<leader>fg", ":FzfLua live_grep<CR>", { desc = "fzf grep" }),
      vim.api.nvim_set_keymap("n", "<leader>fb", ":FzfLua buffers<CR>", { desc = "fzf buffers" }),
      vim.api.nvim_set_keymap("n", "<leader>fh", ":FzfLua helptags<CR>", { desc = "fzf helptags" }),
    })
  end,
```

**Reply by ibhagwan** [reply]
> Hello, on the opposite end I want to have the header, especially for live_grep, but when setting it up like so, I have the header appear on everything else except live_grep, not sure what I am doing wrong:


You’re setting the header string globally on fzf, fzf-lua should respect that, I personally wouldn’t do that and instead have the alt-i|h to appear, open a feature request and I’ll add that :)

**Reply by bleacheda** [reply]
If it doesn't make sense, no need to implement it :).
Maybe I haven't described my small issue correctly, without any config, in the "live_grep" window I do not have the header suggesting the `alt-h' and 'alt-i' shorcuts, it only shows the "ctrf-g" hint. Is that the default? I dont't know, the "files" picker has the "h" applied by default and no hints at all.

![image](https://github.com/user-attachments/assets/6967e3ca-4f1a-40aa-bafe-42b555d95180)
![image](https://github.com/user-attachments/assets/4e35d613-ac16-4966-944d-225f8386f826)

**Reply by ibhagwan** [reply]
> If it doesn't make sense, no need to implement it :).
> Maybe I haven't described my small issue correctly, without any config, in the "live_grep" window I do not have the header suggesting the `alt-h' and 'alt-i' shorcuts, it only shows the "ctrf-g" hint. Is that the default? I dont't know, the "files" picker has the "h" applied by default and no hints at all.

you explained it fine, the headers are disabled by default as it would be too long to display all 3 actions, if you really want these shown post your feedback in #1795 and I’ll add an option when I get a chance.
